packages/fluentui/react-bindings/src/hooks/useFirstMount.ts

```diff
@@ -0,0 +1,19 @@
+import * as React from 'react';
+
+/**
+ * Checks if components was mounted the first time.
+ * @example
+ * const isFirstMount = useFirstMount();
+ *
+ * @returns {Boolean}
+ */
+export function useFirstMount(): boolean {
```

Comment:
why don't you use this hook in `useCallbackRef` ? you can also toggle the value using a layout effect here

Reply:
Good question, it's related to implementation of this in `useCallbackRef` as it's a bit different.

```tsx
// in useCallbackRef.ts
// it's a ref, so will read an actual state
const isFirst = React.useRef(true);

// from useFirstMount() we will get a boolean
const isFirst = useFirstMount();

/* ------------------------ */

// in useCallbackRef.ts, this `useState()` will create an object with closure that will be never updated
const [ref] = React.useState(() => ({
  set current(value) {
    // ✅  uses ref, will use an actual value
    if (skipInitialResolve && isFirst.current) {
      return;
    }

    // 🟥 will stuck with "true" forever as it was in scope initially
    if (skipInitialResolve && isFirst) {
      return;
    }
  }
}));
```

A showcase on CodeSandbox: https://codesandbox.io/s/dreamy-curran-odqsz?file=/src/App.js

Reply:
makes sense to have a ref object in this case, thx